### DISCUSSION TITLE
Shared Global Data- GetStaticProps?

### DISCUSSION THREAD

**Discussion by okaforc-w** [vercel/next.js#24618] [Help]
Hello, I'm very interested in finding out the situation regarding globalising the getStaticProps in the main _app component. At the moment, for example, i created a context api for sharing product data for my ecommerce website. It currently has static product data which is fetched from my firebase database (unless i redeploy with new products). Because the data is static, I want to have it on my index page and also on, individual product page as well as the storefront page. The data is static and can be fetched just once so I definitely dont want it be serverside rendered because thats just much slower. The solution that seems logical is to fetch the data once in the main _app component and then store it in my context api and use it throughout the app. The solution for this seems like only a getInitialProps which makes my entiire application serverSide rendered. Because I definitely dont want that for the end user to request this data each time, I've resulted to some duplication of code for each page requiring the data to use getStaticProps to pre-render these pages at the moment. I've been hearing for more than a year and a half now, that the Next team have the intention to implement this getStaticProps on the main _app component page but there is nothing as of yet. Is there any update on this or is there an efficient way around get shared static props througout the application please

**Comment by okaforc-w** [comment]
anyone?

**Comment by makovez** [comment]
IT IS ABSOLUTELY CRAZY. 

This is most far one of the most important thing and the team doesen't litteraly give a f****. Im loosing days and days finding a NORMAL way to do it, but it doesen't seems to exist. 

I'm really considering to trash away all of my project written in nextjs and choose other framework.

**Reply by makovez** [reply]
btw im in your same situation. Did u find any solution? What you think of react-query with cache? It's ok but I want it to be built at build time as it is totally static.

**Reply by okaforcj** [reply]
Unfortunately not, It's actually really frustrating not to have this seemingly easy but important feature. I wish I could pay for the feature to be made available but I'm not getting anywhere with all the forums. I use firebase and so I cache my data server side but honestly, that is still not good enough I dont think. I'll just have to keep waiting cos the nextJs team have done a really good job with everything else  @soermejo

**Comment by Vadorequest** [comment]
See https://github.com/vercel/next.js/issues/10933#issuecomment-849181620
